presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -265,14 +266,181 @@ public JdbcTableHandle getTableHandle(JdbcIdentity identity, SchemaTableName sch
     public ConnectorSplitSource getSplits(JdbcIdentity identity, JdbcTableLayoutHandle layoutHandle)
     {
         JdbcTableHandle tableHandle = layoutHandle.getTable();
-        JdbcSplit jdbcSplit = new JdbcSplit(
-                connectorId,
-                tableHandle.getCatalogName(),
-                tableHandle.getSchemaName(),
-                tableHandle.getTableName(),
-                layoutHandle.getTupleDomain(),
-                layoutHandle.getAdditionalPredicate());
-        return new FixedSplitSource(ImmutableList.of(jdbcSplit));
+        List<JdbcSplit.RangeInfo> ranges = null;
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            DatabaseMetaData metaData = connection.getMetaData();
+            // determine a column to split the whole table scan
+            ColumnInfo splitColumn = getSplitColumn(metaData, tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName());
+
+            if (splitColumn != null) {
+                log.info("Table: %s.%s.%s, splitColumn: %s", tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName(), splitColumn);
+
+                // collect the min/max stats for the specified split column
+                String minMaxSql = format("select min(%s) AS MIN_VALUE, max(%s) AS MAX_VALUE from %s",
```

Comment:
Not really a big fan of doing this. This poses
- strong assumption on uniform distribution for types other than `auto_increment` and
- introducing of coordinator overhead on getting metadata. This might be a minor issue given most other connectors would do something similar.

Reply:
Yeah, there is an assumption here. Do you have other recommendation for collecting the stats?

Reply:
For the formatting issue, will check the doc, I thought I would be enough once it passed the maven checkstyle plugin.

Reply:
- You could abstract out something like a `SplitSuppiler` with two implementations. One is the existing 1 split per instance and one is yours. Use a config to decide which implementation to inject/use.
- There are many more things checkstyle won't capture: "Avoid abbreviations or acronyms in names", etc. Just go through the formatting guideline to make sure your patch follows the guideline.